src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java

```diff
@@ -2003,6 +2003,14 @@ public void addTo(Ref.IdentityCollection identities)
 
     }
 
+    public boolean isPresent(DecoratedKey key)
+    {
+        // if we don't have bloom filter(bf_fp_chance=1.0 or filter file is missing),
+        // we check index file instead.
+        IFilter bf = getBloomFilter();
```

Comment:
```
     [echo] Running Eclipse Code Analysis.  Output logged to /.../build/ecj/eclipse_compiler_checks.txt
     [java] ----------
     [java] 1. ERROR in /.../src/java/org/apache/cassandra/io/sstable/format/SSTableReader.java (at line 2011)
     [java]     return bf instanceof AlwaysPresentFilter && getPosition(key, Operator.EQ, false) != null || bf.isPresent(key);
     [java]     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
     [java] Potential resource leak: 'bf' may not be closed at this location
```

can probably just avoid `getBloomFilter()` here

Reply:
adding `eclipse-warnings` to my scripts so I run this more often!